docs/src/meshes/p4est_mesh.md

```diff
@@ -12,3 +12,224 @@ such as the compressible Euler equations can use [`FluxRotated`](@ref) to
+   physical domain. The mapping is approximated using interpolation polynomial of a user
+   specified polynomial degree. The default value of this polynomial degree is `1` that
+   corresponds to an uncurved geometry.
+2. High-order boundary information is available in the `.inp` mesh file because it was created
+   with the [`HOHQMesh`](https://github.com/trixi-framework/HOHQMesh.jl) generator.
+   This information is used to create appropriate transfinite mappings during the mesh construction.
+
+We divide our discussion into two parts. The first part discusses the standard node and element information
+contained in the `.inp` mesh file. The second part specifically deals with the mesh file parsing of an Abaqus
+file created by `HOHQMesh`.
+
+### Mesh file header
+
+An Abaqus `.inp` mesh file typically begins with a `*Heading`.
+Though *optional*, the `*Heading` is helpful to give users some information about the mesh described by the mesh file.
+In particular, a `.inp` mesh file created with `HOHQMesh` will contain the header
+```
+*Heading
+ File created by HOHQMesh
+```
+This heading is used to indicate to the mesh constructor which of the above mapping strategies to apply in order to
+create a curvilinear mesh.
+If the Abaqus file header is **not** present then the `P4estMesh` is created with the first strategy above.
+
+### List of corner nodes
+
+Next, prefaced with `*NODE`, comes a list of the physical `(x,y,z)` coordinates of all the corners.
+The first integer in the list of the corner nodes provides the node id number.
+Thus, for the two-dimensional example mesh this block of corner node information is
+```
+*NODE
+1, 1.0, -1.0, 0.0
+2, 3.0,  0.0, 0.0
+3, 1.0,  1.0, 0.0
+4, 2.0,  0.0, 0.0
+5, 0.0,  0.0, 0.0
+6, 3.0,  1.0, 0.0
+7, 3.0, -1.0, 0.0
+```
+
+### List of elements
+
+The element connectivity is given after the list of corner nodes. The header for this information block is
+```
+*ELEMENT, type=CPS4, ELSET=Surface1
+```
+The Abaqus element type `CPS4` corresponds to a quadrilateral element.
+Each quadrilateral element in the unstructured mesh is dictated by four corner points with indexing
+taken from the numbering given by the corner list above.
+The elements connect a set of four corner points (starting from the bottom left) in an anti-clockwise fashion;
+making the element *right-handed*.
+This element handedness is indicated using the circular arrow in the figure above.
+Just as with the corner list, the first integer in the element connectivity list indicates the element id number.
+Thus, the element connectivity list for the three element example mesh is
+```
+*ELEMENT, type=CPS4, ELSET=Surface1
+1, 5, 1, 4, 3
+2, 4, 2, 6, 3
+3, 7, 2, 4, 1
+```
+
+*As a short note, in a three-dimensional Abaqus file the element connectivity would be given in terms of the
+eight corner nodes that define a hexahedron. Also, the header of the element section changes to be*
```

Comment:
Is this sufficient information to determine in which order the nodes are given? I mean, in 2D right-handedness is intuitively clear to me, but is there a unique equivalent in 3D and should we mention it here?

Reply:
There is (somewhat) you basically do the 2d ordering for the bottom 4 nodes and then do a similar ordering for the top 4 nodes. This is how HOHQMesh and p4est both number their corners.

Reply:
OK, thanks. But I agree that this can be added when actual 3D support is available.

Reply:
I made a new 3D figure to explain this a bit more in-depth and will add to the docs. Stay tuned.